graspologic/plot/plot.py

```diff
@@ -671,6 +674,248 @@ def pairplot(
+    cluster_palette : str or dict, optional, default: dictionary using 'Set1'
+        with no input : dictionary using 'Set1'
+    title : string
+        default : "Fit a Gaussian mixture using " + n_components + " components"
+    context :  None, or one of {talk (default), paper, notebook, poster}
+        Seaborn plotting context
+    font_scale : float, optional, default: 1
+        Separate scaling factor to independently scale the size of the font
+        elements.
+    alpha : float, optional, default: 0.7
+        Opacity value of plotter markers between 0 and 1
+    """
+    # Handle X
+    if not isinstance(X, (list, np.ndarray)):
+        msg = "X must be array-like, not {}.".format(type(X))
+        raise TypeError(msg)
+
+    # Handle labels
+    if labels is not None:
+        if not isinstance(labels, (list, np.ndarray)):
+            msg = "Y must be array-like or list, not {}.".format(type(labels))
+            raise TypeError(msg)
+        elif X.shape[0] != len(labels):
+            msg = "Expected length {}, but got length {} instead for Y.".format(
+                X.shape[0], len(labels)
+            )
+            raise ValueError(msg)
+    # Handle gmm
+    if gmm is None:
+        msg = "You must input a sklearn.mixture.GaussianMixture"
+        raise NameError(msg)
+    Y_, means, covariances = gmm.predict(X), gmm.means_, gmm.covariances_
+    data = pd.DataFrame(data=X)
+    n_components = gmm.n_components
+    # setting up the data DataFrame
+    if labels is None:
+        lab_names = [i for i in range(n_components)]
+        data["labels"] = np.asarray([lab_names[Y_[i]] for i in range(Y_.shape[0])])
+    else:
+        data["labels"] = labels
+    data["clusters"] = Y_
+    # creating a cluster_palette from a stirng  only if labels is given
+    if isinstance(cluster_palette, str) and labels is not None:
+        colors = sns.color_palette(cluster_palette, n_components)
+        cluster_palette = dict(zip(np.unique(Y_), colors))
+    # creating a label_palette from a stirng  only if labels is given
+    # note that these can be same string and mapping would be same for both
+    if isinstance(label_palette, str) and labels is not None:
+        colors = sns.color_palette(label_palette, n_components)
+        label_palette = dict(zip(np.unique(np.asarray(labels)), colors))
+    # must give labels AND palettes
+    if (labels is not None) and (cluster_palette is None or label_palette is None):
+        msg = "if give labels array must give palette dicts or string desciptors"
+        raise ValueError(msg)
+    elif labels is None and (cluster_palette is not None or label_palette is not None):
+        msg = "must give labels if using palettes for labels and clusters"
+        raise ValueError(msg)
+    # This takes care of default case, when no palettes or labels given
+    elif labels is None and cluster_palette is None and label_palette is None:
+        colors = sns.color_palette("Set1", n_components)
```

Comment:
don't hard-code the palette here - for most stuff in seaborn for instance, if you pass in a string as `palette` then that string is used in the color palette function, or something like that.

Reply:
@bdpedigo  I do need to this because my plot_ellipse function is expecting a dictionary of labels to colors. Is that alright?

Reply:
you can make the color dictionary yourself when a sting is passed in as the `palette`? Does that make sense?

Reply:
@bdpedigo yes I agree, but in the default case here i am using the palette "set1" to do this. I am using the string to make a dictionary?

Reply:
the default can be a string instead of none? im not sure i understand the issue

Reply:
https://github.com/microsoft/graspologic/pull/541/commits/5e10e21da53efaa8390e55e641c6183d2a0cd11d yes you are right! and it looks neater!